dio/lib/src/dio/dio_for_native.dart

```diff
@@ -140,6 +140,11 @@ class DioForNative with DioMixin implements Dio {
         }).catchError((Object e) async {
           try {
             await subscription.cancel();
+            closed = true;
+            await raf.close();
+            if (deleteOnError && file.existsSync()) {
+              await file.delete();
```

Comment:
This operation should be the same:
```suggestion
              await file.delete().catchError((_) => file);
```